include/libtorrent/lazy_entry.hpp

```diff
@@ -354,6 +354,9 @@ namespace libtorrent {
 			swap(m_len, e.m_len);
 		}
 
+		lazy_entry(lazy_entry&&) = default;
+		lazy_entry& operator=(lazy_entry&&) = default;
```

Comment:
could you explain this?

Reply:
Due to explicitly declared constructor and copy assignment (private), move constructor and move assignment will not be generated automatically. Hence the  declaration.

Reply:
> I think after you move-from I think you need to call release(). I think there should be a test of moving a lazy_entry as well, to make sure there aren't any double-frees

Agree and will add a test.

Reply:
right, I'm suggesting that the automatically generated move constructor and move assignment may be incorrect